force-app/main/default/lwc/jobSearchResult/jobSearchResult.js

```diff
@@ -0,0 +1,296 @@
+
+const ACTIONS = [
+    { label: 'Save As Job Application', name: 'save' }
+];
+
+const COLS = [
+    {
+        label: 'Title', 
+        fieldName: TITLE_FIELD.fieldApiName, 
+        type: 'text', 
+        wrapText: true, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        label: 'Company', 
+        fieldName: COMPANY_FIELD.fieldApiName, 
+        type: 'text', 
+        wrapText: true, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        label: 'Location', 
+        fieldName: LOCATION_FIELD.fieldApiName, 
+        type: 'text', 
+        wrapText: true, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        label: 'Type', 
+        fieldName: TYPE_FIELD.fieldApiName, 
+        type: 'text', 
+        wrapText: true, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        label: 'Salary', 
+        fieldName: SALARY_FIELD.fieldApiName, 
+        type: 'text', 
+        wrapText: true, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        label: 'Link', 
+        fieldName: LINK_FIELD.fieldApiName, 
+        type: 'url', 
+        wrapText: false, 
+        editable: false, 
+        displayReadOnlyIcon: true, 
+        sortable: true, 
+        cellAttributes: { alignment: 'left' }
+    }, 
+    {
+        type: 'action', 
+        typeAttributes: { rowActions: ACTIONS, menuAlignment: 'auto' }
+    }
+];
+
+export default class JobSearchResult extends LightningElement {
+    columns = COLS;
+
+    defaultSortDirection = 'asc';
+    sortDirection = 'asc';
+    sortedBy;
+
+    recordIds = [];
+
+    @wire(queryJobs) jobs;
+
+    handleSort(event) {
+        const { fieldName: sortedBy, sortDirection } = event.detail;
+        const CLONE_DATA = [...this.jobs.data];
+
+        CLONE_DATA.sort(this.sortBy(sortedBy, sortDirection === 'asc' ? 1 : -1));
+
+        this.jobs.data = CLONE_DATA;
+        this.sortDirection = sortDirection;
+        this.sortedBy = sortedBy;
+    }
+
+    sortBy(field, reverse, primer) {
+        const KEY = primer 
+            ? function (x) { return primer(x[field]); }
+            : function (x) { return x[field]; };
+
+        return function (a, b) {
+            a = KEY(a);
+            b = KEY(b);
+
+            return reverse * ((a > b) - (b > a));
+        };
+    }
+
+    async handleRowAction(event) {
+        const ACTION = event.detail.action;
+        const ROW = event.detail.row;
+
+        switch (ACTION.name) {
```

Comment:
I don't know that a switch is necessary since you are only handling a single action